### PR TITLE
String gate sanity checks

### PR DESCRIPTION
**Pull request by Python1320**
Add sane maximum length check for certain operations. The system cannot handle arbitrarily big operations.

These sanity checks should be extended to other constructs such as array and E2 strings also.

### CODE REVIEW

File: lua/wire/gates/string.lua [hunk 1]
```diff
@@ -131,13 +133,23 @@ GateActions["string_find"] = {
...
+		if  (A and #A or 0) 
+		  + (B and #B or 0) 
+		  + (C and #C or 0) 
+		  + (D and #D or 0) 
+		  + (E and #E or 0) 
+		  + (F and #F or 0) 
+		  + (G and #G or 0) 
+		  + (H and #H or 0)  > MAX_LEN   
+		then
+			return false
```
Comment (CaptainPRICE) [inline lua/wire/gates/string.lua]: Why `return false`, shouldn't this return an empty string instead?
(Same on line 270)

Comment (dvdvideo1234) [inline lua/wire/gates/string.lua]: I think is should return an empty string as table.Concat does

Comment (AbigailBuccaneer) [inline lua/wire/gates/string.lua]: Yeah, if the gate is supposed to output a string then it should actually return a string (unless there's some special handling of incorrectly-typed values that presents errors in a graceful way).

Comment (Divran) [inline lua/wire/gates/string.lua]: I checked the code and the base wire gate entity will convert "nil" or "false" into the zero value of whatever type it's supposed to be. So I'm merging this now.

**Pull request merged**